search/includes/classes/class-versioning.php

```diff
@@ -200,27 +201,68 @@ public function get_inactive_versions( Indexable $indexable ) {
 	 * @return array Array of index versions
 	 */
 	public function get_versions( Indexable $indexable, bool $provide_default = true ) {
-		$versions = get_option( self::INDEX_VERSIONS_OPTION, array() );
+		$versions = [];
+
+		if ( $indexable->global ) {
+			$versions = get_site_option( self::INDEX_VERSIONS_OPTION_GLOBAL, array() );
+		} else {
+			$versions = get_option( self::INDEX_VERSIONS_OPTION, array() );
+		}
 
 		$slug = $indexable->slug;
 
-		if ( ! isset( $versions[ $slug ] ) || ! is_array( $versions[ $slug ] ) || empty( $versions[ $slug ] ) ) {
-			if ( $provide_default ) {
-				return array(
-					1 => array(
-						'number' => 1,
-						'active' => true,
-						'created_time' => null, // We don't know when it was actually created
-						'activated_time' => null,
-					),
+		if ( ! $this->_get_versions_is_slug_defined( $versions, $slug ) ) {
+			// Check deprecated location
+			$deprecated_versions = get_site_option( self::INDEX_VERSIONS_OPTION, array() );
```

Comment:
Technically, the deprecated location would have only been used if the site is in network mode. To prevent unnecessary option lookups here, should we put this behind the same network mode check that update_versions() used to use?

Reply:
Good idea. Will do.

Reply:
Done